ui/AddManager.php

```diff
@@ -153,62 +169,52 @@ private function addManagerSortFn($a, $b) {
         return (substr($_REQUEST["sortBy"], -1, 1) == "-")?-$retval:$retval;
     }
     
-    private function addManagerColHeader($subaction, $header, $static) {
-        $command = $header;
-        if(!strcmp($header, $_REQUEST["sortBy"])) {
-            $command .= "-";
-            $selected = 1;
-        } else if(!strcmp($header . "-", $_REQUEST["sortBy"]))
-            $selected = 2;
-    
-        if($header == "Num") $width = " WIDTH=60px";
-    
-        if($header == "%") $header = "&nbsp;%&nbsp;";
-    
-        if($static)
-            echo "        <TH ALIGN=LEFT$width><U>$header</U>";
-        else
-            echo "        <TH ALIGN=LEFT$width><A CLASS=\"nav\" HREF=\"?session=".$this->session->getSessionID()."&amp;action=addmgr&amp;subaction=$subaction&amp;sortBy=$command&amp;date=".$_REQUEST["date"]."\">$header</A>";
-    
-        if($selected && !$static)
-            echo "&nbsp;<SPAN CLASS=\"sort" . (($selected==1)?"Down":"Up") . "\"><IMG SRC=\"img/blank.gif\" WIDTH=8 HEIGHT=4 ALT=\"\"></SPAN>";
-    
-        // Give extra horizontal padding to column
-        ////if(!strcmp($header, "Num"))
-        ////    echo "&nbsp;&nbsp;&nbsp;";
-    
-        echo "</TH>\n";
+    private function getAlbumCategory($albumRow, $categoryMap) {
+       $category = null;
+       $acats = explode(",", $albumRow["afile_category"]);
+       foreach($acats as $index => $cat) {
+           if($cat)
+               $category = $categoryMap[$cat-1]["code"];
+       }
+
+       $category = $category == null ? "G" : $category;
+       return $category;
     }
-    
+
+    private function getEditCell($albumRow) {
+       $sessionId="?session=".$this->session->getSessionID();
+
+       $hrefDate= $sessionId .  "&amp;action=addmgr&amp;subaction=adds&amp;date=$date ";
+       $class = ($id && $id == $row["id"]) ? "sel" : "nav";
+       $cellDate = "<A CLASS='nav' HREF='" . $hrefDate .
+          "' onClick='ConfirmDelete(".$row["id"]."); return false;'>[x]</A>&nbsp;";
+
+       $hrefId =  $sessionId . "&amp;action=addmgr&amp;subaction=addsedit&amp;id=" . $row["id"];
+       $cellId = "<A CLASS='songEdit' HREF='" . $hrefId . "'>&#x270f;</A>";
+
+       return "<TD>" . $cellDate . $cellId . "</TD>";
+    }
+
```

Comment:
works for me and the html is well formed running both logged in and anonymous modes. are you running in some special case?

Reply:
Hello Eric, no special case.  In the HTML, the ID is blank in the HREF.

Looking at the method getEditCell, there are a couple variables $row and $id.  These are not defined in this method.  Is this the code in your local environment, because I cannot see how it would actually work at all.

Reply:
fixed. ids not included in links. please test because it is not clear what these actions should actually do.

Reply:
done.